pandas/tests/indexes/test_base.py

```diff
@@ -76,16 +76,13 @@ def test_constructor_casting(self, index):
         tm.assert_contains_all(arr, new_index)
         tm.assert_index_equal(index, new_index)
 
-    @pytest.mark.xfail(
-        using_string_dtype() and not HAS_PYARROW, reason="TODO(infer_string)"
-    )
     def test_constructor_copy(self, using_infer_string):
         index = Index(list("abc"), name="name")
         arr = np.array(index)
         new_index = Index(arr, copy=True, name="name")
         assert isinstance(new_index, Index)
         assert new_index.name == "name"
-        if using_infer_string:
+        if using_infer_string and HAS_PYARROW:
             tm.assert_extension_array_equal(
                 new_index.values, pd.array(arr, dtype="str")
             )
```

Comment:
This one I don't understand. Why is `index.values` a numpy array in case of "python" storage?

Reply:
Yeah I thought I rolled that one back, the change was just something I tried.

Updated now